### DISCUSSION TITLE
Question: common feature between MParT and TransportMaps libraries

### DISCUSSION THREAD

**Discussion by vlopezmar** [MeasureTransport/MParT#316] [Q&A] [answered]
Thank you for making your measure transport libraries available!

I have been using the TransportMaps library and I wanted to know whether the MParT library has functionality equivalent to that from the 
    map_samples_base_to_target()   and   map_samples_target_to_base() functions from the pull back and push forward distributions available from the TransportMaps library.
(https://transportmaps.mit.edu/docs/api-TransportMaps-Distributions.html#TransportMaps.Distributions.PullBackTransportMapDistribution)

Thanks again.

**Comment by dannys4** [comment, marked as answer]
Thanks for the question!
Currently there is a pull request open (#305 ) that will allow you to train a map in the binding languages based on an objective, where the current implementation requires you to use the KL-divergence from the Gaussian as the standalone example that's currently bound. Feel free to look in the test in `bindings/python/test/TestTrainMap.py` (I believe) for how this is used in Python to create a map from samples to transport to a standard normal measure.

Obviously, this isn't the only measure you'd be interested in targeting, and hopefully we'll have time moving forward to start exposing more power to the user to use `TrainMap` with different interesting objectives and KL divergences with respect to different distributions. For now, if you have a situation where you'd like to create a map with some other kind of objective, or use an arbitrary target distribution, you can take a look at our examples repository at [MeasureTransport/MParT-examples](https://github.com/MeasureTransport/MParT-examples), and their rendered Jupyter notebooks [on the documentation website](https://measuretransport.github.io/MParT/source/tutorials/index.html)

**Reply by vlopezmar** [reply]
Thank you very much for the pointers.  I had already gone through the examples -- that is what got me interested :-) . I will look at the bindings directory as you suggest.  In any case, I will keep following the developments.

Thanks again!
